packages/opentelemetry-core/src/distribute-context/propagation/HttpTraceContext.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Carrier,
+  Context,
+  HttpTextFormat,
+  DistributedContext,
+} from '@opentelemetry/api';
+
+import {
+  getDistributedContext,
+  setDistributedContext,
+} from '../distribute-context';
+
+export const CORRELATION_CONTEXT_HEADER = 'Correlation-Context';
+
+/**
+ * Propagates {@link DistributedContext} through Context format propagation.
+ *
+ * Based on the Correlation Context specification:
+ * https://w3c.github.io/correlation-context/
+ */
+export class HttpTraceContext implements HttpTextFormat {
+  inject(context: Context, carrier: Carrier) {
+    const distContext = getDistributedContext(context);
+    if (distContext) {
+      const headerValue = Object.keys(distContext)
+        .map(
+          (key: string) =>
+            `${encodeURIComponent(key)}=${encodeURIComponent(
```

Comment:
I think we should consider below limits as per the specs: https://w3c.github.io/correlation-context/#header-value

Limits:

- Maximum number of name-value pairs: 180.
- Maximum number of bytes per a single name-value pair: 4096.
- Maximum total length of all name-value pairs: 8192.

Reply:
+1, I'll d those limits.

Reply:
Should we throw an exeption when the limits are excedeed? or skip the key-value pair? WDYT?

Reply:
In general we never throw if we can avoid it. I would suggest to log and skip WDYT @mayurkale22 

This may be a question for spec as well.